File: Rock Paper Scissors.py
import random
import time
import sys

#Lists the variables.
choice = 0
optionlist = ("rock","paper","scissors")
option = random.choice(optionlist)
again = "yes"

#Start of the script.
print("Welcome to rock paper scissors.")
time.sleep(3)
#Loops over and over as long as the user wants to play.
while again == "yes":
    #Computer randomly picks an option and user picks an option.
    option = random.choice(optionlist)
    choice = input("What is your choice? \n(rock/paper/scissors) \n")
    print(f"I picked {option}.")
    time.sleep(3)
    #Compares the choices made and works out the winner.
    if choice == "rock":
        if option == "rock":
            print("It's a draw...")
        elif option == "paper":
            print("Yes, I win!")
        elif option == "scissors":
            print("There's always next time.")
    elif choice == "paper":
        if option == "rock":
            print("Which means you win...")
        elif option == "paper":
            print("Therefore we draw.")
        elif option == "scissors":
            print("I win!")
    elif choice == "scissors":
        if option == "rock":
            print("You lose!")
        elif option == "paper":
            print("You win...")
        elif option == "scissors":
            print("Hmm a draw.")
    time.sleep(4)
    #Asks the user if they want to play again.
    again = input("Would you like to go again? \n(yes/no) \n")
input("ok then, press enter to exit \n")
sys.exit()

File: RockPaperScissors.java
import java.util.Random;
import java.util.Scanner;

class RPS {
    public static void main(String[] args) {
        String choice;
        String[] optionList = {"rock", "paper", "scissors"};
        Scanner scn = new Scanner(System.in);
        String option;
        Random randIndex = new Random();
        String again = "yes";

        System.out.println("Welcome to rock paper scissors.");
        while (again.equals("yes")) {
            option = optionList[randIndex.nextInt(3)];
            System.out.println("What is your choice? \r\n(rock/paper/scissors)");
            choice = scn.nextLine();
            System.out.println("I picked " + option + ".");
            if (choice.equals("rock")) {
                if (option.equals("rock")) {
                    System.out.println("It's a draw.");
                } else if (option.equals("paper")) {
                    System.out.println("I win!");
                } else if (option.equals("scissors")) {
                    System.out.println("There's always next time...");
                }
            } else if (choice.equals("paper")) {
                if (option.equals("paper")) {
                    System.out.println("It's a draw.");
                } else if (option.equals("scissors")) {
                    System.out.println("I win!");
                } else if (option.equals("rock")) {
                    System.out.println("There's always next time...");
                }
            } else if (choice.equals("scissors")) {
                if (option.equals("scissors")) {
                    System.out.println("It's a draw.");
                } else if (option.equals("rock")) {
                    System.out.println("I win!");
                } else if (option.equals("paper")) {
                    System.out.println("There's always next time...");
                }
            }
            System.out.println("Would you like to play again? \r\n(yes/no)");
            again = scn.nextLine();
        }
        scn.close();
    }
}